frontend/src/components/tables/usage.js

```diff
@@ -0,0 +1,200 @@
+import { PageSection, Card, CardBody, Title } from "@patternfly/react-core";
+import { ChartDonut } from "@patternfly/react-charts";
+
+import ConnectionError from "../error";
+import Preloader from "../preloader";
+import { useQuery } from "react-query";
+
+const UsageComponent = (props) => {
+    const fetchData = () =>
+        fetch("http://127.0.0.1:5000/api/usage/" + props.what).then(
+            (response) => response.json(),
+        );
+
+    const { data, isLoading, isError } = useQuery(
+        "usage" + props.what,
+        fetchData,
+        {
+            keepPreviousData: true,
+        },
+    );
```

Comment:
Looking good! I would change it up a bit so that the `fetchData` isn't tied to the component, which also helps with performance. Overall changing the `what` prop to granularity might also be a good idea, as `what` doesn't tell much.
```suggestion
const fetchDataByGranularity = (granularity) =>
    fetch("http://127.0.0.1:5000/api/usage/" + granularity).then(
        (response) => response.json(),
    );
const UsageComponent = (props) => {
    const { data, isLoading, isError } = useQuery(
        "usage" + props.what,
        () => fetchData(props.what),
        {
            keepPreviousData: true,
        },
    );
```

Reply:
Thanks!